### PR TITLE
DIV-6228: Citizen respondent co-respondent can submit AOS response whilst bailiff application in progress

### PR DESCRIPTION
**Pull request by madjava**
# Description
https://tools.hmcts.net/jira/browse/DIV-6228

Updating RFE to enable respondent or co-respondent respond, progress with Aos even after the bailiff process has began. Cases that are in any of the bailiff related states can still either be linked or responded to by this update.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by yarpo — APPROVED** [review]

File: core/utils/petitionHelper.js [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
+  proceedButDisagree: 'proceedButDisagree',
+  defend: 'defend',
+  yes: 'Yes',
+  no: 'No',
+  notAccept: 'NoNoAdmission',
+  behavior: 'unreasonable-behaviour',
+  desertion: 'desertion',
+  separation5yrs: 'separation-5-years',
+  coRespondent: 'correspondent'
+};
```
Comment (jacmurphy) [inline core/utils/petitionHelper.js:23]: image

File: mocks/services/case-orchestration/retrieve-aos-case/GET_Authorization=Bearer coRespNotDefendingBailiffCase.mock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+
+#import './mock-coRespNotDefendingBailiffCase.json';
```
Comment (jacmurphy) [inline mocks/services/case-orchestration/retrieve-aos-case/GET_Authorization=Bearer coRespNotDefendingBailiffCase.mock:4]: Is this needed ?  There seems to be a number of files with this addition or similar. Could you enlighten me on what its used for ?

**Review by johnoneill46 — APPROVED** [review]

File: mocks/services/case-orchestration/retrieve-aos-case/mock-adultery-defendedIssuedToBailiff.json [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    "jurisdictionConfidentLegal": "Yes",
+    "petitionerContactDetailsConfidential": "share",
+    "petitionerFirstName": "BANANA",
+    "petitionerLastName": "ban",
+    "respondentFirstName": "Apple",
+    "respondentLastName": "App",
+    "marriagePetitionerName": "Ban",
+    "marriageRespondentName": "App",
+    "petitionerNameDifferentToMarriageCertificate": "No",
+    "petitionerEmail": "vivdivred@mailinator.com",
```
Comment (jacmurphy) [inline mocks/services/case-orchestration/retrieve-aos-case/mock-adultery-defendedIssuedToBailiff.json:47]: Quick question. 
Recently IDAM seem to have noticed as a collective whole, HMCTS have generated a large amount of email addresses for testing. 

Could this email add to that problem? 
If so, is there an email we can highlight an email as whitelisted for the use of our tests (that wont get deleted by IDAM) that will make sure we wont have to do a large number of changes to what email is used? @yarpo @madjava

**Review by jacmurphy — CHANGES_REQUESTED** [review]
Could you please inform me on what the following additions are for ?

mocks/services/case-orchestration/retrieve-aos-case/GET_Authorization=Bearer coRespNotDefendingBailiffCase.mock

mocks/services/case-orchestration/retrieve-aos-case/GET_Authorization=Bearer yesCaseDefendedBailiffStarted.mock

mocks/services/case-orchestration/retrieve-aos-case/mock-adultery-defendedIssuedToBailiff.json

File: mocks/services/case-orchestration/retrieve-aos-case/mock-adultery-defendedIssuedToBailiff.json (hunk 3 above)
Comment (madjava) [reply mocks/services/case-orchestration/retrieve-aos-case/mock-adultery-defendedIssuedToBailiff.json:47]: @jacmurphy These email, and others on the mock data don't have any impact, they are simply just dummy data. The ones you are referring to are the user details created in the e2e or integration test.

**Comment by madjava** [comment]
@jacmurphy Those are for mock screen, for local dev and test and it basically mocks calls to COS with the expected response in the json file

File: mocks/services/case-orchestration/retrieve-aos-case/GET_Authorization=Bearer coRespNotDefendingBailiffCase.mock (hunk 2 above)
Comment (madjava) [reply mocks/services/case-orchestration/retrieve-aos-case/GET_Authorization=Bearer coRespNotDefendingBailiffCase.mock:4]: These mock calls to COS and return data when testing, working locally.

File: charts/div-rfe/values.preview.template.yaml [hunk 4]
```diff
@@ -9,6 +9,7 @@ nodejs:
         LOG_LEVEL: "DEBUG"
         REDISCLOUD_URL: "redis://${SERVICE_NAME}-redis-master:6379"
         EQUALITY_URL: "https://pcq.aat.platform.hmcts.net"
+        COS_BASE_URL: "http://div-cos-pr-1351.service.core-compute-preview.internal"
```
Comment (madjava) [inline charts/div-rfe/values.preview.template.yaml:12]: Would be reverted before merge. In here to test on COS PR

**Review by jacmurphy — APPROVED** [review]

**Pull request merged**